solidstart-tanstackquery-tailwind-modules/src/components/PRAndIssuesListItem/PRAndIssuesListItem.tsx

```diff
@@ -34,36 +37,29 @@ const PRAndIssuesListItem = (props: Issue) => {
         <div class="flex-shrink-0 pl-4 translate-y-1">
           <span
             class={cn({
-              'text-green-600': local.state === 'OPEN',
-              'text-gray-500': local.state !== 'OPEN' && type() === 'issue',
+              'text-green-600': item?.state === 'OPEN',
+              'text-gray-500': item?.state !== 'OPEN' && type === 'issue',
               'text-purple-600':
-                (local.state !== 'OPEN' && local.state === 'MERGED') ||
-                (local.state === 'CLOSED' && type() === 'issue'),
-              'text-red-600': type() === 'pr' && local.state === 'CLOSED',
+                (item?.state !== 'OPEN' && item?.state === 'MERGED') ||
+                (item?.state === 'CLOSED' && type === 'issue'),
+              'text-red-600': type === 'pr' && item?.state === 'CLOSED',
             })}
           >
-            <Show when={type() === 'issue'}>
-              <Switch>
-                <Match when={local.state === 'OPEN'}>
-                  <IssuesIcon class="w-5 h-5" />
-                </Match>
-                <Match when={local.state === 'CLOSED'}>
-                  <ClosedIssueIcon class="w-5 h-5" />
-                </Match>
-              </Switch>
+            <Show when={item?.state === 'OPEN' && type === 'issue'}>
```

Comment:
Not sure how you feel about it, but personally I would extract all this Icon code into a component that took `type` and `state`

```tsx
const iconComponents = {
  issue: {
    OPEN: IssuesIcon,
    CLOSED: ClosedIssueIcon,
  },
  pr: {
    OPEN: PullRequestIcon,
    CLOSED: ClosedPrIcon,
    MERGED: MergedPrIcon,
  },
} as const;

const Icon = (props: {
  type: 'issue' | 'pr';
  state: Props['item']['state'];
}) => {
  const component = createMemo(
    () =>
      Object.entries(iconComponents[props.type]).find(
        ([key]) => key === props.state
      )?.[1]
  );

  const classes = createMemo(() =>
    cn('w-5 h-5', {
      'text-green-600': props.state === 'OPEN',
      'text-gray-500': props.state !== 'OPEN' && props.type === 'issue',
      'text-purple-600':
        (props.state !== 'OPEN' && props.state === 'MERGED') ||
        (props.state === 'CLOSED' && props.type === 'issue'),
      'text-red-600': props.type === 'pr' && props.state === 'CLOSED',
    })
  );

  return <Dynamic component={component()} class={classes()} />;
};
```

then just have:
```tsx
<div class="flex-shrink-0 pl-4 translate-y-1">
  <Icon type={props.type} state={props.item.state} />
</div>
```